snp_matrix.pl

```diff
@@ -0,0 +1,147 @@
+#!/usr/bin/env perl
+use strict;
+use File::Basename;
+use Bio::AlignIO;
+use Bio::SimpleAlign;
+use Bio::LocatableSeq;
+use Cwd;
+use Getopt::Long;
+
+my $script_name = $0;
+
+sub usage
+{
+	"Usage: ".basename($script_name)." [pseudoalign.phy]\n".
+	"Constructs a snp matrix from the pseudoalignment file of the pipeline\n".
+	"Options:\n".
+	"\t-v:  Print more information (to stderr)\n".
+	"\t-o:  Prints matrix to passed file\n";
+}
+
+my $verbose = 0;
+my $output = undef;
+if (not GetOptions('v' => \$verbose, 'o|output=s' => \$output))
+{
+	die "Invalid options\n".usage;
+}
+
+my $input_file = $ARGV[0];
+
+die "Invalid file\n".usage if (not defined $input_file);
+die "File $input_file does not exist\n".usage if (not -e $input_file);
+die "File $input_file is not readable\n".usage if (not -r $input_file);
+die "Invalid file $input_file\n".usage if (-d $input_file);
+
+my $out_h;
+
+if (defined $output)
+{
+	open($out_h,">$output") or die "Could not open $output for writing\n".usage;
```

Comment:
Change open to use the 3 arguments instead of 2.  http://modernperlbooks.com/mt/2010/04/three-arg-open-migrating-to-modern-perl.html